src/config.hpp

```diff
@@ -634,6 +634,13 @@ class config
 	const config& find_mandatory_child(config_key_type key, const std::string &name,
 		const std::string &value) const;
 
+	optional_config_impl<config> find_child_recursive(config_key_type key, const std::string &name,
```

Comment:
Why do you need a non-const version?

Reply:
Was just following the convention in that file, but I technically don't need the non-const version.

Reply:
I double-checked the implementation of `optional_config_impl`, and it looks like it _is_ an optional reference, so that probably is the reason to have two.

Reply:
If we can use one, use one. Using `const_cast` is really not optimal, because there's no guarantee that the const version won't have side effects (which it shouldn't).

Reply:
But if we want to return a reference, we probably need both.

Reply:
> Using `const_cast` is really not optimal,

Will removing the `const_cast` but keeping that function work?
By the same logic, any other functions using `const_cast` should be adjusted as well.

Reply:
Rewrote `find_child_recursive` to not use `const_cast`.

Reply:
Personally I think const_cast is a perfectly legitimate way to handle these kinds of function pairs, where one returns a const reference and the other returns a non-const reference, and the latter uses const_cast. I'd certainly prefer that to writing the entire function twice.

Reply:
It could probably be addressed with a macro or maybe templates? const_cast is something to avoid if at all possible just generally is my understanding.

Reply:
Also yeah, could just remove the non-const version.

Reply:
I think this kind of situation is _precisely_ one of the cases that are an exception to that rule.